src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalyzer.cs

```diff
@@ -12,6 +15,23 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     /// </summary>
     public abstract class DiagnosticAnalyzer
     {
+        /// <summary>
+        /// This field caches the result of <see cref="AnalysisContext.ConfigureGeneratedCodeAnalysis"/> for the first
+        /// initialization containing at least one registered callback.
+        /// </summary>
+        /// <value>
+        /// The value is a <see cref="GeneratedCodeAnalysisFlags"/> stored as an integer, or -1 if the value has not
+        /// been assigned.
+        /// </value>
+        private int _generatedCodeAnalysisFlags = -1;
```

Comment:
Interesting approach to track the value directly on the analyzer. Seems very reasonable for tracking state which will hardly ever change. We have large number of immutable state tracking maps for analyzers in the analyzer driver which get initialized once up front, wondering if we can move more such state directly into the analyzer and reduce dictionary accesses.